File: Data Structures & Algorithms/subsets-ii/submission-0.cpp
class Solution {
public:
    void solve(vector<int> &nums, vector<int> res, vector<vector<int>> & ans, int i){
        if(i == nums.size()){
            ans.push_back(res);
            return;
        }
        res.push_back(nums[i]);
        solve(nums,res, ans, i+1);
        res.pop_back();
        while(i+1 < nums.size() && nums[i] == nums[i+1]){
            i++;
        }
        solve(nums,res,ans, i+1);
        return;
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> res;
        sort(nums.begin(), nums.end());
        solve(nums, res, ans, 0);
        return ans;
    }
};

File: Data Structures & Algorithms/counting-bits/submission-0.java
class Solution {
    public int[] countBits(int n) {
        int [] ans = new int [n+1];
        for(int i = 0 ; i<= n; i++){
            int cnt = 0;
            int j = i;
            while(j > 0){
                if((j& 1) == 1){
                    cnt++;
                }
                j = j>>1;
            }
            ans[i] = cnt;
        }
        return ans;
    }
}
